packages/client/hmi-client/src/components/models/tera-model-configuration.vue

```diff
@@ -122,56 +148,140 @@ import TabView from 'primevue/tabview';
 import TabPanel from 'primevue/tabpanel';
 import InputText from 'primevue/inputtext';
-import { Model } from '@/types/Types';
-import { ModelConfig } from '@/types/ModelConfig';
+import { ModelConfiguration } from '@/types/Types';
+import { AnyValueMap } from '@/types/common';
+import {
+	createModelConfiguration,
+	updateModelConfiguration
+} from '@/services/model-configurations';
 import { useOpenedWorkflowNodeStore } from '@/stores/opened-workflow-node';
-import { NumericValueMap } from '@/types/common';
+import { ModelOperation } from '@/components/workflow/model-operation';
+
+const openedWorkflowNodeStore = useOpenedWorkflowNodeStore();
 
 const props = defineProps<{
-	model: Model;
 	isEditable: boolean;
-	modelConfigNodeInput?: ModelConfig;
+	modelConfigurations: ModelConfiguration[];
 	calibrationConfig?: boolean;
 }>();
 
-const modelConfigNames = ref(['Config 1']);
+const editableModelConfigs = ref<ModelConfiguration[]>([]);
 
 const selectedModelConfig = ref();
 const fakeExtractions = ref(['Resource 1', 'Resource 2', 'Resource 3']);
 
-const initialValues = ref<NumericValueMap[]>([{}]);
-const parameterValues = ref<NumericValueMap[]>([{}]);
 const openValueConfig = ref(false);
 const cellValueToEdit = ref({ data: {}, field: '', index: 0 });
 
+// const
+
 // Selected columns
 const selectedInitials = ref<string[]>([]);
 const selectedParameters = ref<string[]>([]);
 
-const openedWorkflowNodeStore = useOpenedWorkflowNodeStore();
+// const variableTypes = computed(() => props.model.semantics ? Object.keys(props.model.semantics.ode) : []);
 
-const modelConfiguration = computed(() =>
-	modelConfigNames.value.map((name, i) => ({
-		name,
-		...initialValues.value[i],
-		...parameterValues.value[i]
-	}))
+const configurations = computed(
+	() => editableModelConfigs.value?.map((m) => m.configuration.semantics.ode) ?? null
 );
 
+// TODO: Clean this up and use appropriate loops
+const modelConfigurationTable = computed(() => {
+	if (editableModelConfigs.value && configurations.value) {
+		console.log('Configuration', configurations.value);
+
+		const odes: object[] = [];
+
+		for (let i = 0; i < configurations.value.length; i++) {
+			if (configurations.value[i]) {
+				odes.push({});
+				// eslint-disable-next-line
+				for (const key of Object.keys(configurations.value[i])) {
+					odes[i][key] = [];
+
+					configurations.value[i][key].forEach((value) => {
+						const newPair = {};
+						newPair[value.target ?? value.id] = value.expression ?? value.value;
+						odes[i][key].push(newPair);
+					});
+				}
+			}
+		}
+		console.log(odes);
+
+		const variables: AnyValueMap[] = [];
+		// eslint-disable-next-line
+		for (let i = 0; i < odes.length; i++) {
+			variables[i] = {};
+			// eslint-disable-next-line
+			for (const [key, values] of Object.entries(odes[i])) {
+				const flattenedObj = {};
+
+				if (isArray(values)) {
+					// @ts-ignore
+					// eslint-disable-next-line
+					for (let j = 0; j < values.length; j++) {
```

Comment:
This looks like a O(n^3) loop which gives me pause :D

Reply:
ODE is actually an object with a "fixed-number" of keys so it is more like `O(N^2)`.  But the parsing could be clearer and better, perhaps this is something we can flag with a FIXME and we can work offline @shawnyama to improve - the modelConfiguration change is holding back other PRs so I think we don't want to dwell too much time unless it is a critical error.


See example here: https://github.com/DARPA-ASKEM/Model-Representations/blob/main/petrinet/examples/sir.json